### PR TITLE
feat(tables): crud for table resource

### LINKED ISSUE #19: CRUD Table
- [x] Create table
- [x] Read table
- [x] Update table
- [x] Disable table

### PR DESCRIPTION
**Pull request by jcmexdev**
Crud for table resource 
- Create
- Update
- FindAll
- FindOne
## Changes
- Rename Dtos to has the same structure name like others entities 
- Change verification query to update table
- Add status field to table

### CODE REVIEW

File: src/modules/tables/dto/create-table.dto.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+export class CreateTableDto {
+  @ApiProperty({ description: 'The table name', required: true })
+  @IsString({ message: 'table_name_required' })
+  name: string;
+  @ApiProperty({
+    description: 'The table active-inactive status',
+    required: true,
+  })
+  @IsDefined({ message: 'table_is_active_attribute_required' })
```
Comment (jlazo) [inline src/modules/tables/dto/create-table.dto.ts]: Why do you define the attribute isActive? When you create a table is always active, dont you?

File: src/modules/tables/table.entity.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+@Entity('tables')
+export class Table extends BaseEntity {
+  @PrimaryGeneratedColumn('increment')
+  id: number;
+  @Column({ type: 'varchar', name: 'name', length: 30 })
+  name: string;
+  @Column({
+    type: 'boolean',
+    name: 'is_active',
+    default: false,
```
Comment (jlazo) [inline src/modules/tables/table.entity.ts]: The default value must be true

File: src/modules/tables/tables.controller.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+} from '@nestjs/common';
+import { TablesService } from './tables.service';
+import { CreateTableDto } from './dto/create-table.dto';
+import { UpdateTableDto } from './dto/update-table.dto';
+import { ApiOperation, ApiTags } from '@nestjs/swagger';
+import { AuthGuard } from '@nestjs/passport';
+
+@ApiTags('The tables')
+@Controller('tables')
+export class TablesController {
```
Comment (jlazo) [inline src/modules/tables/tables.controller.ts]: Why is the module name  plural and not singular like the rest?

File: src/modules/tables/tables.service.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    return plainToClass(ResponseTableDto, table);
+  }
+
+  async update(id: number, updateTableDto: UpdateTableDto) {
+    const table = await this._tableRepository.findOne(id);
+
+    if (!table) throw new ConflictException('table_was_not_found');
+
+    //Query to get a table with the name to update to compare if the name is already taken
+    const storedTable = await this._tableRepository.findOne({
```
Comment (jlazo) [inline src/modules/tables/tables.service.ts]: Podrias hacer un query que busque por nombre y donde el id sea diferente para ahorrarle el if que viene abajo

**Review by jlazo — CHANGES_REQUESTED** [review]

File: src/modules/tables/tables.controller.ts (hunk 3 above)
Comment (jcmexdev) [reply src/modules/tables/tables.controller.ts]: You are right i'm going to rename it.

File: src/modules/tables/tables.service.ts (hunk 4 above)
Comment (jcmexdev) [reply src/modules/tables/tables.service.ts]: Tienes toda la razón, no se me ocurrió eso, lo cambiare 😎🔥

File: src/modules/tables/table.entity.ts (hunk 2 above)
Comment (jcmexdev) [reply src/modules/tables/table.entity.ts]: Yes, let me change it  🤖

File: src/modules/tables/dto/create-table.dto.ts (hunk 1 above)
Comment (jcmexdev) [reply src/modules/tables/dto/create-table.dto.ts]: Ok, I think that it is reasonable i'm going to change it

File: src/modules/table/table.service.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+  async findOne(id: number): Promise<TableResponseDto> {
+    const table = await this._tableRepository.findOne(id);
+
+    if (!table) throw new ConflictException('table_was_not_found');
+
+    return plainToClass(TableResponseDto, table);
+  }
+
+  async update(id: number, TableUpdateDto: TableUpdateDto) {
```
Comment (jlazo) [inline src/modules/table/table.service.ts]: The param mus be camel case

File: src/modules/table/table.entity.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  UpdateDateColumn,
+} from 'typeorm';
+
+@Entity('tables')
+export class Table extends BaseEntity {
+  @PrimaryGeneratedColumn('increment')
+  id: number;
+  @Column({ type: 'varchar', name: 'name', length: 30 })
+  name: string;
+  @Column({ type: 'boolean', name: 'status', default: true })
```
Comment (jlazo) [inline src/modules/table/table.entity.ts]: Use a more descriptive name as is_available o is_free

File: src/modules/table/table.controller.ts [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+import { TableService } from './table.service';
+import { TableCreateDto } from './dto/table-create.dto';
+import { TableUpdateDto } from './dto/table-update.dto';
+import { ApiOperation, ApiTags } from '@nestjs/swagger';
+import { AuthGuard } from '@nestjs/passport';
+
+@ApiTags('The tables')
+@Controller('tables')
+export class TableController {
+  constructor(private readonly _TableService: TableService) {}
```
Comment (jlazo) [inline src/modules/table/table.controller.ts]: replace _TableService by _tableService

**Review by jlazo — CHANGES_REQUESTED** [review]

**Pull request closed without merging**